sidestep_engine/core/run_discovery.py

```
@@ -356,19 +356,66 @@ def load_run_config(
+    ]
+    ckpt_root = run_dir / "checkpoints"
+    if ckpt_root.is_dir():
+        # Pick the highest-numbered epoch dir that has the file
+        epoch_dirs = sorted(
+            (d for d in ckpt_root.iterdir()
+             if d.is_dir() and d.name.startswith("epoch_")),
+            key=lambda d: parse_epoch_num(d.name),
+            reverse=True,
+        )
+        for ed in epoch_dirs:
+            ac_candidates.append(ed / "sidestep_adapter_config.json")
```

Ruff (0.15.7)

[warning] Use `list.extend` to create a transformed list

(PERF401)